core/server/monkey-patch.js

```diff
@@ -0,0 +1,28 @@
+var TimezonedDate = require('timezoned-date'),
+    lodash = require('lodash'),
+    clonedeep = require('lodash.clonedeep'),
+    toString = require('lodash.tostring');
+
+// ######## LODASH
+// timezoned-date does not work with lodash.cloneDeep 3.x
+lodash.cloneDeep = clonedeep;
+
+// @TODO: reason?
+lodash.toString = toString;
+
+// ######## NATIVE JS DATE
+// remember original timezone offset of server
+global.ServerTimezoneOffset = new Date().getTimezoneOffset();
+
+// timezoned-date is able to force UTC at process level
+// we only use it when local system is NOT UTC
+if (ServerTimezoneOffset !== 0) {
```

Comment:
What I'm saying is that `new Date().getTimezoneOffset() === 0` doesn't mean there in no offset. It just means there is no offset **now**. But it can appear some time later. You might start your server a minute before the clocks switch from 0 to +1.

As for the migration script, are you sure `moment-timezone` can't help here? It seems to know all the DST switch dates for all the time zones. If you know the original TZ of those dates, what prevents you from converting them to UTC with moment?